src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js

```diff
@@ -0,0 +1,303 @@
+      expect(container.getByText(messageIds.loanable)).toBeInTheDocument();
+    });
+  });
+
+  describe('Render when "isLoanable" returns true', () => {
+    let container;
+
+    beforeEach(() => {
+      container = renderLoansSection({
+        ...initialProps,
+        policy: {
+          ...mockedPolicy,
+          isLoanable: jest.fn(() => true),
+        }
+      });
+    });
+
+    it('Should render loan profile dropdown correctly', () => {
+      const expectedProps = {
+        component: Select,
+        id: 'input_loan_profile',
+        name: 'loansPolicy.profileId',
+      };
+
+      expect(Field).toHaveBeenNthCalledWith(2, expect.objectContaining(expectedProps), {});
+      expect(container.getByText(messageIds.loanProfile)).toBeInTheDocument();
+    });
+
+    it('Should render due date dropdown correctly', () => {
+      const expectedProps = {
+        component: Select,
+        name: 'loansPolicy.closedLibraryDueDateManagementId',
+        children: {
+          options: longTermLoansOptions,
+          id: messageIds.pleaseSelect,
+        }
+      };
+
+      expect(Field).toHaveBeenNthCalledWith(3, expect.objectContaining(expectedProps), {});
+      expect(container.getByText(messageIds.dueDate)).toBeInTheDocument();
+    });
+
+    it('Should render item limit input correctly', () => {
+      const expectedProps = {
+        component: TextField,
+        name: 'loansPolicy.itemLimit',
+        id: 'input_item_limit',
+        type: 'number',
+      };
+
+      expect(Field).toHaveBeenNthCalledWith(4, expect.objectContaining(expectedProps), {});
+      expect(container.getByText(messageIds.itemLimit)).toBeInTheDocument();
+    });
+
+    it('"Period" component should get correct properties under "Grace period" section', () => {
+      const expectedProps = {
+        fieldLabel: messageIds.gracePeriod,
+        inputValuePath: 'loansPolicy.gracePeriod.duration',
+        selectValuePath: 'loansPolicy.gracePeriod.intervalId',
+        changeFormValue: initialProps.change,
+      };
+
+      expect(Period).toHaveBeenCalledWith(expect.objectContaining(expectedProps), {});
+    });
+  });
+
+  describe('Render when "isProfileRolling" returns true', () => {
+    let container;
+
+    beforeEach(() => {
+      container = renderLoansSection({
+        ...initialProps,
+        policy: {
+          ...mockedPolicy,
+          isProfileRolling: jest.fn(() => true),
+        }
+      });
+    });
+
+    it('"Period" component should get correct properties under "Loan period" section', () => {
+      const expectedProps = {
+        fieldLabel: messageIds.loanPeriod,
+        inputValuePath: 'loansPolicy.period.duration',
+        selectValuePath: 'loansPolicy.period.intervalId',
+        changeFormValue: initialProps.change,
+      };
+
+      expect(Period).toHaveBeenCalledWith(expect.objectContaining(expectedProps), {});
+    });
+
+    it('Should render due date schedule dropdown correctly', () => {
```

Comment:
Also we can add `expect` about `schedules`.
You can add `data-testid` for `Field` component and try to find mocked schedules inside.

Reply:
It is not necessary to add `data-testid` attribute. I can check visibility of `schedules` without it

Reply:
Ofcourse, but with `data-testid` you can do something like:
`expect(within(screen.getByTestId('yourTestId')).getByText('yourScheduleText')).toBeVisible()`
and will be shure that your `schedules` situated inside of correct component.
But it's up to you.

Reply:
checked as children property instead